File: array/two_sum/two_sum.py
from typing import List


def two_sum(nums: List[int], target: int)->List[int]:
    dic = {}
    for i, num in enumerate(nums):
        if num in dic:
            return [dic[num], i]
        else:
            dic[target - num] = i


if __name__ == "__main__":
    arr = [3, 2, 4]
    target = 6
    res = two_sum(arr, target)
    print(res)

File: array/two_sum/two_sum.js
function twoSum(nums, target) {
    const dic = new Map()
    for (let i = 0; i < nums.length; i++) {
        const num = nums[i]
        if (dic.has(num)) {
            return [dic.get(num), i]
        }
            dic.set(target - num, i)

    }
}

const arr = [3, 2, 4]
const target = 6
const res = twoSum(arr, target)
console.log(res)
